src/lib/merkle_ledger/null_ledger.ml

```diff
@@ -0,0 +1,129 @@
+open Core_kernel
+
+module Make
+    (Key : Intf.Key)
+    (Account : Intf.Account with type key := Key.t)
+    (Hash : Intf.Hash with type account := Account.t)
+    (Location : Location_intf.S) (Depth : sig
+        val depth : int
+    end) : sig
+  include
+    Base_ledger_intf.S
+    with module Addr = Location.Addr
+    with module Location = Location
+    with type key := Key.t
+     and type key_set := Key.Set.t
+     and type hash := Hash.t
+     and type root_hash := Hash.t
+     and type account := Account.t
+
+  val create : unit -> t
+end = struct
+  type t = Core.Uuid.t [@@deriving sexp_of]
+
+  let t_of_sexp _ = failwith "t_of_sexp unimplemented"
+
+  type index = int
+
+  module Location = Location
+  module Path = Merkle_path.Make (Hash)
+
+  type path = Path.t
+
+  module Addr = Location.Addr
+
+  let create () = Core.Uuid.create ()
+
+  let remove_accounts_exn _t = failwith "null ledgers cannot be mutated"
+
+  let empty_hash_at_heights depth =
```

Comment:
Not a big deal, but the hashes never change here, so you could turn this into a static array from 0..depth-1 of pre-computed hashes.

Reply:
Although it does depend on the particular `Hash` module.

Reply:
isn't that what `memoized_empty_hash_at_height` does right below?

Reply:
Yep, sorry I didn't notice!